apps/web/app/(navigation)/EditBudget.tsx

```diff
@@ -0,0 +1,243 @@
+"use client";
+
+import * as Tabs from "@radix-ui/react-tabs";
+import { Form, Field, FormInstance } from "houseform";
+import { useTranslate } from "lib/hooks";
+import { useValidateBudgetModal } from "./useValidateBudgetModal";
+import { Budget } from "lib/types";
+import { Modal, IconPicker, Input, CustomDatePicker, Button } from "ui";
+import {
+  SeparatorStyledTop,
+  TabsStyled,
+  TabsTriggerStyled,
+  ContentStyled,
+  ParagraphStyled,
+  IconPickerStyled,
+  InputWrapperFullStyled,
+  TextAreaWrapperStyled,
+  TextareaStyled,
+  TextareaErrorStyled,
+  InputWrapperFullFlex,
+  DatePickerErrorStyled,
+  DatePickerWrapperStyled,
+  SeparatorStyled,
+  ButtonWrapperStyled,
+} from "./CreateNewBudget.styled";
+
+import { icons } from "./CreateNewBudget";
+interface EditBudgetProps {
+  budget: Budget;
+  handleHideEditBudgetModal: () => void;
+  onClose: () => void;
+  handleBudgetsEdit: (budgetsAfterEdit: Budget[]) => void;
+}
+
+export const EditBudget = ({
+  budget,
+  onClose,
+  handleHideEditBudgetModal,
+  handleBudgetsEdit,
+}: EditBudgetProps) => {
+  const { t, dict } = useTranslate("EditBudgetModal");
+
+  const { checkNameOnChange, checkNameOnSubmit, checkDescription, checkDate } =
+    useValidateBudgetModal("AddNewBudgetModal");
+
+  const defaultValueTabs = "settings";
+
+  const getDateObject = (dateType: string) => {
+    if (dateType === "start-date") {
+      return budget.startDate ? new Date(budget.startDate) : null;
+    } else if (dateType === "end-date")
+      return budget.endDate ? new Date(budget.endDate) : null;
+  };
+
+  const getSelectedValue = (value: Date | null) =>
+    value ? new Date(value) : null;
+
+  const checkEndDateOnChangeValidate = (
+    val: Date | null,
+    form: FormInstance<any>
+  ) => {
+    const start = val! && val.getTime();
+    const end =
+      form.getFieldValue("start-date")!.value &&
+      form.getFieldValue("start-date")!.value.getTime();
+
+    if (start && end) {
+      if (start < end) return Promise.reject(t(dict.errors.dateBeforeStart));
+    }
+    return Promise.resolve(true);
+  };
```

Comment:
I think that there are a few things here that are a little confusing, because of the variables' names used 👀 

For example, there is `const end` but it regards `start-date`, and `start` variable is really the timestamp of end date. As a result, because of the current names of the variables, the "if" statement: `if (start < end)` at first glance makes you wonder why there is an error for a situation when `start < end`.

I also think that you can simplify code for `start` and `end` variables using `?` operator.

To sum up, after changing variables' names and mentioned above simplifaction you could arrive at this:

```suggestion
  const checkEndDateOnChangeValidate = (
    endDateValue: Date | null,
    form: FormInstance<any>
  ) => {
    const endTimestamp = endDateValue?.getTime();
    const startTimestamp = form.getFieldValue("start-date")!.value?.getTime();

    if (endTimestamp && startTimestamp) {
      if (startTimestamp > endTimestamp) return Promise.reject(t(dict.errors.dateBeforeStart));
    }
    return Promise.resolve(true);
  };
```